### ISSUE TITLE
ImportError: libnetcdf.so.13

### ISSUE THREAD

**Issue by msuruzhon** [michellab/BioSimSpace#151]
Hi there,

I am mentioning this issue here, because it seems to be related to BioSimSpace/Sire imports:

https://github.com/protocaller/ProtoCaller/issues/8

It seems that some dependencies are incomplete? Note that this is BioSimSpace 2019.2 and this information could be outdated.

I appreciate the feedback.

**Comment by chryswoods** [comment]
Thanks for posting. @lohedges will take a look soon. My guess is that a conda install for ProtoCaller has uninstalled the netcdf library. This happens a lot with conda... Could be our error if we haven't properly marked netcdf as a dependency, but conda does do some very strange things when other packages are installed.

**Comment by lohedges** [comment]
Hi @msuruzhon. Just to check, are you installing using a clean conda environment?

Sire is built using the `netcd4` package from `conda-forge`, which installs `libnetcdf`. The version of `netcdf4` that is used is pinned in the conda recipe (template [here](https://github.com/michellab/Sire/blob/devel/docker/sire-conda/recipe/template.yaml)) at build time using the scripts in [this](https://github.com/michellab/Sire/tree/devel/docker/sire-conda) directory. Issues can occur when other packages within your conda environment (or dependencies of Sire/BioSimSpace) happen to change the installed version of `netcd4f` behind your back.

What I imagine has happened is that another package, e.g. `rdkit`, might have pulled in a newer `netcdf4` while conda somehow still thinks things are compatible. To debug, could you try installing just our Sire conda package within a clean environment and seeing if that works. Then try BioSimSpace, or adding the BioSimSpace deps one-by-one, to see if that breaks thing. If BioSimSpace works, try adding the deps for ProtoCaller and seeing what happens.

Cheers.

**Comment by lohedges** [comment]
It's also good to know if this is a Linux or macOS issue since the conda packages are often updated out of sync or have inconsistent dependencies and recipes.

**Comment by msuruzhon** [comment]
Thanks for the helpful replies! This could be indeed the case, i.e. conda doing something strange either because of an existing environment, or because of weird dependency relationships.

It seems that it is a Linux machine but I don't know any more specifics, so I am tagging the OP:
@niko97320

**Comment by chryswoods** [comment]
We've joked in the team that we can probably write a book about all of the various issues with handling and packaging python dependencies. It is so brittle...

**Comment by lohedges** [comment]
Hi again,

I've just tested on my own Linux box and installing ProtoCaller as per your GitHub instructions works fine for me. (Using a clean conda environment.) It installs `libnetcdf-4.6.2` from `conda-forge` and I end up with `libnetcdf.so.13` in the lib directory of the conda environment.

I can only assume that @niko97320 has done a non-standard install, or used a non-clean environment.

(I think version pinning in Sire first appeared in 2019.2.0 since I noticed lots of issues with conda's `pin_compatible` directive. We still can't deal with dependencies of dependencies though, which can change without your knowledge.)
